pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -19,7 +19,34 @@
 
 let
   version = "246";
-in stdenv.mkDerivation {
+
+  libudev = systemd.overrideAttrs (_: {
+    pname = "libudev";
+
+    outputs = [ "out" "dev" ];
+
+    buildInputs =
+      [ linuxHeaders libcap curl.dev kmod xz pam acl
+        libuuid glib libgcrypt libgpgerror libidn2
+        pcre2 ] ++
+        stdenv.lib.optional withKexectools kexectools ++
+        stdenv.lib.optional withLibseccomp libseccomp ++
+      [ libffi audit lz4 bzip2 libapparmor
+        iptables gnu-efi
+      ] ++ stdenv.lib.optional withSelinux libselinux;
```

Comment:
Most of these probably are not dependencies of just libudev, are they?

Reply:
Yes, I was planning to do such reductions **after** the change actually [works](https://github.com/NixOS/nixpkgs/pull/97051#issuecomment-686786509).

Reply:
Now I tried to start with empty `mkDerivation` instead of overriding the systemd one, but I couldn't come with anything simple.  Just making the meson configure pass isn't easy &ndash; it requires lots of stuff that's useless for libudev but I see no easy way of turning these off (`*Inputs`, `mesonFlags`, etc.) &ndash; honestly it feels that maintainability will be easiest when those two derivations only have minimal differences (full systemd and libudev).